mne/beamformer/_lcmv.py

```diff
@@ -102,52 +133,86 @@ def _apply_lcmv(data, info, tmin, forward, noise_cov, data_cov, reg,
+            # finding optimal orientation for NAI and unit-noise gain
+            # based on [2]_, Eq. 4.47
+            tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+            eig_vals, eig_vecs = linalg.eig(np.dot(linalg.pinv(tmp),
+                                                   np.dot(Wk, Gk)))
+            idx_max = eig_vals.argmax()
+            max_ori = eig_vecs[:, idx_max]
+            Wk[:] = np.dot(max_ori, Wk)
+            Gk = np.dot(Gk, max_ori)
 
-            # TODO: The eigenvector associated with the smallest eigenvalue
-            # should probably be used when using combined EEG and MEG data
-            max_ori = eig_vecs[:, idx_middle]
+            # compute spatial filter for NAI
+            tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+            denom = np.sqrt(tmp)
+            Wk /= denom
+            if weight_norm == 'nai':
+                Wk /= np.sqrt(noise)
 
-            Wk[:] = np.dot(max_ori, Wk)
-            Ck = np.dot(max_ori, np.dot(Ck, max_ori))
             is_free_ori = False
 
-        if is_free_ori:
-            # Free source orientation
-            Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
+        # NAI without orientation selection
+        elif weight_norm == 'nai' and pick_ori != 'max-power':
+            raise ValueError('Weight normalization with neural activity index '
+                             'is not implemented yet with free or fixed '
+                             'orientation.')
+
+        # all other combinations are handled here:
         else:
-            # Fixed source orientation
-            Wk /= Ck
+            if pick_ori == 'max-power':
+                eig_vals, eig_vecs = linalg.eigh(Ck)
+
+                # Choosing the eigenvector associated with the middle
+                # eigenvalue. The middle and not the minimal eigenvalue is used
+                # because MEG is insensitive to one (radial) of the three
+                # dipole orientations and therefore the smallest eigenvalue
+                # reflects mostly noise.
+                for ii in range(3):
+                    if ii != eig_vals.argmax() and ii != eig_vals.argmin():
+                        idx_middle = ii
+
+                # TODO: The eigenvector associated with the smallest eigenvalue
+                # should probably be used when using combined EEG and MEG data
+                max_ori = eig_vecs[:, idx_middle]
+
+                Wk[:] = np.dot(max_ori, Wk)
+                Ck = np.dot(max_ori, np.dot(Ck, max_ori))
+                is_free_ori = False
+
+            if is_free_ori:
+                # Free source orientation
+                Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
+            else:
+                # Fixed source orientation
+                Wk /= Ck
 
     # Pick source orientation maximizing output source power
     if pick_ori == 'max-power':
         W = W[0::3]
+    else:
+        if weight_norm is not None:
```

Comment:
weight-norm = 'nai' and orientation selection other than max-power results in an error in line 159. but it's probably clearer to use 'unit-noise-gain' here?